lib/interjectable.rb

```diff
@@ -45,18 +55,29 @@ def inject(dependency, &default_block)
     #     @@dependency ||= instance_eval(&default_block)
     #   end
     def inject_static(dependency, &default_block)
-      cvar_name = "@@#{dependency}"
-      remove_class_variable(cvar_name) if class_variable_defined?(cvar_name)
+      raise MethodAlreadyDefined if instance_methods(false).include?(dependency)
```

Comment:
Ditto about including a the method name in a useful error message

Reply:
Good call.